lib/components/result-view/index.js

```diff
@@ -1,46 +1,82 @@
 /* @flow */
 
-import { CompositeDisposable } from "atom";
+import { CompositeDisposable, Point } from "atom";
 import React from "react";
 
 import { reactFactory } from "./../../utils";
+import OutputStore from "./../../store/output";
 import ResultViewComponent from "./result-view";
 
-import type OutputStore from "./../../store/output";
+import type MarkerStore from "./../../store/markers";
 import type Kernel from "./../../kernel";
 
 export default class ResultView {
-  element: HTMLElement;
   disposer: atom$CompositeDisposable;
   marker: atom$Marker;
 
   destroy = () => {
     this.disposer.dispose();
-    if (this.marker) this.marker.destroy();
+    this.marker.destroy();
   };
 
   constructor(
-    store: OutputStore,
-    kernel: Kernel,
-    marker: atom$Marker,
+    markerStore: MarkerStore,
+    kernel: ?Kernel,
+    editor: atom$TextEditor,
+    row: number,
     showResult: boolean = true
   ) {
-    this.marker = marker;
-    this.element = document.createElement("div");
-    this.element.classList.add("hydrogen", "marker");
+    const element = document.createElement("div");
+    element.classList.add("hydrogen", "marker");
 
     this.disposer = new CompositeDisposable();
 
+    markerStore.clearOnRow(row);
+
+    const buffer = editor.getBuffer();
+    const lineLength = buffer.lineLengthForRow(row);
+
+    const point = new Point(row, lineLength);
+    this.marker = editor.markBufferPosition(point, { invalidate: "touch" });
+    const lineHeight = editor.getLineHeightInPixels();
+
+    const outputStore = new OutputStore();
+    outputStore.updatePosition({
+      lineLength: lineLength,
+      lineHeight: editor.getLineHeightInPixels(),
+      editorWidth: editor.getEditorWidthInChars()
+    });
+
+    editor.decorateMarker(this.marker, {
+      type: "block",
+      item: element,
+      position: "after"
+    });
+
+    this.marker.onDidChange(event => {
+      if (!event.isValid) {
+        markerStore.delete(this.marker.id);
+      } else {
+        outputStore.updatePosition({
+          lineLength: this.marker.getStartBufferPosition().column
+        });
+      }
+    });
+
+    markerStore.new(this);
+
     reactFactory(
       <ResultViewComponent
-        store={store}
+        store={outputStore}
         kernel={kernel}
         destroy={this.destroy}
         showResult={showResult}
       />,
-      this.element,
+      element,
       null,
       this.disposer
     );
+
+    return outputStore;
   }
```

Comment:
I want expect new return it's instance(instance of `ResultView`). 
If caller want to access outputStore, I like this even if it's verbose.( so need to keep outputStore as prop of `ResultView`)

```javascript
const resultView = new ResultView(...)
const {outputStore} = resultView
```